### PR TITLE
Initial appeals status API endpoint

### PR DESCRIPTION
**Pull request by shanear**
connects #1930 

## Test plan

- Automated tests: Integration API test & unit tests

- Manual test with VACOLS data, example response:
```
{
    "data": [
        {
            "attributes": {
                "active": false,
                "events": [
                    {
                        "date": "2007-03-27",
                        "type": "nod"
                    },
                    {
                        "date": "2007-04-25",
                        "type": "soc"
                    },
                    {
                        "date": "2007-05-10",
                        "type": "form9"
                    },
                    {
                        "date": "2008-01-31",
                        "type": "certified"
                    },
                    {
                        "date": "2010-11-09",
                        "type": "bva_final_decision"
                    }
                ],
                "prior_decision_date": null,
                "type": "original"
            },
            "id": "1773852",
            "type": "appeals"
        },
        {
            "attributes": {
                "active": false,
                "events": [
                    {
                        "date": "2006-07-12",
                        "type": "nod"
                    },
                    {
                        "date": "2007-03-29",
                        "type": "soc"
                    },
                    {
                        "date": "2007-06-30",
                        "type": "withdrawn"
                    }
                ],
                "prior_decision_date": null,
                "type": "original"
            },
            "id": "1685649",
            "type": "appeals"
        },
        {
            "attributes": {
                "active": false,
                "events": [
                    {
                        "date": "2005-08-22",
                        "type": "nod"
                    },
                    {
                        "date": "2006-03-02",
                        "type": "soc"
                    },
                    {
                        "date": "2006-05-31",
                        "type": "withdrawn"
                    }
                ],
                "prior_decision_date": null,
                "type": "original"
            },
            "id": "1564597",
            "type": "appeals"
        },
        {
            "attributes": {
                "active": false,
                "events": [
                    {
                        "date": "2005-01-06",
                        "type": "nod"
                    },
                    {
                        "date": "2006-03-11",
                        "type": "soc"
                    },
                    {
                        "date": "2006-05-31",
                        "type": "withdrawn"
                    }
                ],
                "prior_decision_date": null,
                "type": "original"
            },
            "id": "1472182",
            "type": "appeals"
        },
        {
            "attributes": {
                "active": false,
                "events": [
                    {
                        "date": "2003-09-16",
                        "type": "nod"
                    },
                    {
                        "date": "2004-03-29",
                        "type": "soc"
                    },
                    {
                        "date": "2004-05-31",
                        "type": "withdrawn"
                    }
                ],
                "prior_decision_date": null,
                "type": "original"
            },
            "id": "1302899",
            "type": "appeals"
        }
    ]
}
```

### CODE REVIEW

File: app/controllers/api/v1/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Api::V1::ApplicationController < ActionController::Base
```
Comment (shanear) [inline app/controllers/api/v1/application_controller.rb]: The `ActionController` for the API needed almost nothing from the main `ActionController`, so I spun a new one off here.

File: app/models/appeal_event.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
+class AppealEvent
+  include ActiveModel::Model
+
+  # TODO: Confirm this list with Chris, do we have all dispositions accounted for?
```
Comment (shanear) [inline app/models/appeal_event.rb]: @cmgiven I completely guessed on this mapping. Can you take a look?

File: app/models/appeal_events.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  def form9_event
+    AppealEvent.new(type: :form9, date: appeal.form9_date)
+  end
+
+  def ssoc_events
+    appeal.ssoc_dates.map { |ssoc_date| AppealEvent.new(type: :ssoc, date: ssoc_date) }
+  end
+
+  # TODO: Confirm with Chris
+  # if not decision_date, then disposition doesn't matter
```
Comment (shanear) [inline app/models/appeal_events.rb]: @cmgiven just confirming, we will only have a decision event based on the disposition if the `decision_date` is set?

File: app/models/serializers/v1/appeal_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+class V1::AppealSerializer < ActiveModel::Serializer
+  def id
+    object.vacols_id
+  end
+
+  # TODO: Validate with Chris
```
Comment (shanear) [inline app/models/serializers/v1/appeal_serializer.rb]: Another mapping I had no clue how to make, we are missing a lot of possible VACOLS appeal types here.

File: app/models/appeal.rb [hunk 5]
```diff
@@ -309,6 +318,10 @@ def fetch_end_products(vbms_id)
       bgs.get_end_products(vbms_id).map { |ep_hash| EndProduct.from_bgs_hash(ep_hash) }
     end
 
+    def for_veteran_ssn(ssn)
+      repository.appeals_by_veteran_ssn(ssn).sort_by(&:nod_date).reverse
```
Comment (shanear) [inline app/models/appeal.rb]: @cmgiven Ordered appeals by nod_date in reverse chronological order. Sound right?

File: app/services/appeal_repository.rb [hunk 6]
```diff
@@ -98,8 +108,12 @@ def self.set_vacols_values(appeal:, case_record:)
       case_record: case_record,
       disposition: VACOLS::Case::DISPOSITIONS[case_record.bfdc],
       decision_date: normalize_vacols_date(case_record.bfddec),
+      prior_decision_date: normalize_vacols_date(case_record.bfdpdcn),
       status: VACOLS::Case::STATUS[case_record.bfmpro],
-      outcoding_date: normalize_vacols_date(folder_record.tioctime)
+      outcoding_date: normalize_vacols_date(folder_record.tioctime),
+
+      # TODO: Is this the veteran or appellant SSN?
```
Comment (shanear) [inline app/services/appeal_repository.rb]: @cmgiven TODO above is a question for you

File: app/services/appeal_repository.rb [hunk 7]
```diff
@@ -35,6 +35,16 @@ def self.load_vacols_data(appeal)
     return false
   end
 
+  def self.appeals_by_veteran_ssn(ssn)
+    cases = MetricsService.record("VACOLS: appeals_by_veteran_ssn",
+                                  service: :vacols,
+                                  name: "appeals_by_veteran_ssn") do
+      VACOLS::Correspondent.find_by!(ssn: ssn).cases.includes(:folder, :correspondent)
```
Comment (shanear) [inline app/services/appeal_repository.rb]: `find_by!` so we can distinguish if the vet doesn't exist, or if the vet just doesn't have any appeals

File: app/controllers/api/v1/appeals_controller.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  def ssn
+    request.headers["ssn"]
+  end
+
+  def appeals
+    @appeals ||= Appeal.for_veteran_ssn(ssn)
+  end
+
+  def verify_feature_enabled
+    not_found unless FeatureToggle.enabled?(:appeals_status)
```
Comment (shanear) [inline app/controllers/api/v1/appeals_controller.rb]: no the API doesn't have users or logins, that's why i avoided that method.

File: app/controllers/api/v1/application_controller.rb [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  before_action :setup_fakes
+
+  private
+
+  def not_found
+    render json: { status: "not found" }, status: 404
+  end
+
+  def ssl_enabled?
+    Rails.env.production?
```
Comment (shanear) [inline app/controllers/api/v1/application_controller.rb]: weirdly enough, the `Rails.env.production?` is true for `UAT` and all AWS envs lol

File: app/controllers/api/v1/appeals_controller.rb (hunk 8 above)
Comment (aroltsch) [inline app/controllers/api/v1/appeals_controller.rb]: ok, makes sense

File: app/services/appeal_repository.rb (hunk 6 above)
Comment (cmgiven) [inline app/services/appeal_repository.rb]: Appellant.

File: app/models/appeal.rb (hunk 5 above)
Comment (cmgiven) [inline app/models/appeal.rb]: Can we order by the newest event? Remember that appeal streams from the same appeal will all have the same NOD date.

File: app/models/serializers/v1/appeal_serializer.rb (hunk 4 above)
Comment (cmgiven) [inline app/models/serializers/v1/appeal_serializer.rb]: Anything that's not 1, 3, or 7 is out of scope for this version.

File: app/models/appeal_events.rb (hunk 3 above)
Comment (cmgiven) [inline app/models/appeal_events.rb]: Yes.

File: app/models/appeal_event.rb (hunk 2 above)
Comment (cmgiven) [inline app/models/appeal_event.rb]: Yup, I'll take a pass at this one.

Comment (cmgiven) [inline app/models/appeal_event.rb]:
```ruby
DISPOSITIONS_BY_EVENT_TYPE = {
  bva_final_decision: [
    "Allowed",
    "Denied",
    "Vacated",
    "Dismissed, Death",
    "Dismissed, Other"
  ],
  bva_remand: [
    "Remanded",
    "Manlincon Remand"
  ],
  field_grant: [
    "Benefits Granted by AOJ",
    "Advance Allowed in Field"
  ],
  withdrawn: [
    "Withdrawn",
    "Motion to Vacate Withdrawn",
    "Withdrawn from Remand",
    "Recon Motion Withdrawn",
    "Advance Withdrawn Death of Veteran",
    "Advance Withdrawn by Appellant/Rep",
    "Advance Failure to Respond",
    "Remand Failure to Respond"
  ],
  merged: [
    "Merged Appeal"
  ],
  other: [
    "Designation of Record",
    "Reconsideration by Letter"
  ]
}.freeze
```

File: app/models/appeal.rb [hunk 10]
```diff
@@ -338,7 +337,7 @@ def fetch_end_products(vbms_id)
       bgs.get_end_products(vbms_id).map { |ep_hash| EndProduct.from_bgs_hash(ep_hash) }
     end
 
-    def for_api(appellant_ssn: appellant_ssn)
+    def for_api(appellant_ssn:)
```
Comment (aroltsch) [inline app/models/appeal.rb]: Good job Rubocop!!!

**Pull request merged**